analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -215,24 +216,29 @@ variableDeclarator.Parent.Parent switch
                 };
 
             Diagnostic CreateS1144Diagnostic(SyntaxNode syntaxNode, ISymbol symbol) =>
-                Diagnostic.Create(RuleS1144, syntaxNode.GetLocation(), accessibility, symbol.GetClassification(), GetMemberName(symbol));
+                Diagnostic.Create(RuleS1144, GetIssueLocation(syntaxNode), accessibility, symbol.GetClassification(), GetMemberName(symbol));
         }
 
+        private static Location GetIssueLocation(SyntaxNode syntaxNode) =>
```

Comment:
what it does is more like GetIdentifierLocation, right?

also (open question) - do you think it's better like this or would it be better as static local function? (given there's lots of methods in this class and some more organization might help readability)?

Reply:
Couldn't agree more. I turned it into a local static method.